Service factories become backend features (#19334)
backstage/backstage: .changeset/young-lies-attack.md

---
'@backstage/backend-plugin-api': patch
'@backstage/backend-test-utils': patch
'@backstage/backend-defaults': patch
'@backstage/backend-app-api': patch
---

> Removed options from `createBackend`.
> 
> The `createBackend` method doesn't accept any options anymore as in the following:

Reviewer: ```suggestion
**BREAKING**: Removed options from `createBackend`. Service factories are now `BackendFeature`s and should be installed with `backend.add(...)` instead. The following should be migrated:
```

```ts
const backend = createBackend({ services: [myCustomServiceFactory] });
```

> In order to pass custom service factories use the `add` method:

Reviewer: ```suggestion
To instead pass the service factory via `backend.add(...)`:
```
